src/main/java/com/mraof/minestuck/computer/editmode/EditmodeLocations.java

```diff
@@ -0,0 +1,371 @@
+		
+		//TODO consider validating the pos and source
+		if(locations.containsKey(level))
+		{
+			for(Pair<BlockPos, Source> locationPair : locations.get(level))
+			{
+				if(locationPair.getFirst().equals(pos) && locationPair.getSecond().equals(source))
+					return; //prevent duplicates
+			}
+		}
+		
+		locations.put(level, locationPairIn);
+	}
+	
+	public void removeEntry(ResourceKey<Level> level, BlockPos pos, Source source)
+	{
+		if(level == null || pos == null)
+			return;
+		
+		//dont try to remove ENTRY Sources as they are only added once
+		if(source != Source.ENTRY)
+			locations.remove(level, Pair.of(pos, source));
+	}
+	
+	/**
+	 * Takes in a player and looks through each source (such as a computer) providing editmode to see if the player resides within one of the spaces.
+	 * Used both server side and client side
+	 *
+	 * @param editPlayer Player in editmode
+	 * @return Whether the player is standing in a supported region
+	 */
+	public boolean canMoveAtPosition(Player editPlayer, double defaultRange)
+	{
+		Level editLevel = editPlayer.level();
+		double editPosX = editPlayer.getX();
+		double editPosY = editPlayer.getY();
+		double editPosZ = editPlayer.getZ();
+		
+		if(!locations.containsKey(editLevel.dimension()))
+			return false;
+		
+		List<Pair<BlockPos, Source>> posSourcePairs = locations.get(editLevel.dimension()).stream().toList();
+		boolean xInBounds = false;
+		boolean yInBounds = false;
+		boolean zInBounds = false;
+		boolean allMatch = false;
+		boolean noneMatch = true;
+		
+		for(Pair<BlockPos, Source> posSourcePair : posSourcePairs)
+		{
+			//if the source is ENTRY, then it should be 30 regardless of what the config says due to the static nature of the location additions when Entry occurred
+			double moddedRange = posSourcePair.getSecond() == Source.ENTRY ? ENTRY_RANGE : defaultRange;
+			BlockPos iteratePos = posSourcePair.getFirst();
+			
+			int centerX = iteratePos.getX();
+			int centerY = iteratePos.getY();
+			int centerZ = iteratePos.getZ();
+			
+			//if the defaultRange is 1 and the player pos isnt an exact match, dont bother
+			//if(defaultRange == 1 && editPos != iteratePos)
+			//	continue;
+			
+			boolean localXInBounds = isWithinRange(editPosX, centerX, moddedRange);
+			boolean localYInBounds = isWithinRange(editPosY, centerY, moddedRange);
+			boolean localZInBounds = isWithinRange(editPosZ, centerZ, moddedRange);
+			
+			//TODO does not factor in player offset such as in ServerEditHandler
+			if(localXInBounds)
+			{
+				xInBounds = true;
+				noneMatch = false;
+			}
+			
+			if(localYInBounds)
+			{
+				yInBounds = true;
+				noneMatch = false;
+			}
+			
+			if(localZInBounds)
+			{
+				zInBounds = true;
+				noneMatch = false;
+			}
+			
+			if(localXInBounds && localYInBounds && localZInBounds)
+			{
+				allMatch = true;
+				break;
+			}
+		}
+		
+		return handleMovement(editPlayer, xInBounds, yInBounds, zInBounds, allMatch, noneMatch);
```

Comment:
If I understand this loop correctly, you can replace `noneMatch` here with `!xInBounds && !yInBounds && !zInBounds`.